packages/yarnpkg-fslib/tests/npath.test.ts

```diff
@@ -127,6 +157,36 @@ describe(`Portable paths`, () => {
             const outputPath = `C:`;
             expect(npath.fromPortablePath(inputPath)).toEqual(outputPath);
           });
+
+          it(`should transform back UNC Windows paths (/unc/[server]/[sharename]/)`, () => {
+            const inputPath = `/unc/Server01/user/docs/Letter.txt`;
+            const outputPath = '\\\\Server01\\user\\docs\\Letter.txt';
+            expect(npath.fromPortablePath(inputPath)).toEqual(outputPath);
+          });
+
+          it(`should transform back Long UNC Windows paths (/unc/?/[server]/[sharename]/)`, () => {
+            const inputPath = `/unc/?/Server01/user/docs/Letter.txt`;
+            const outputPath = '\\\\?\\Server01\\user\\docs\\Letter.txt';
+            expect(npath.fromPortablePath(inputPath)).toEqual(outputPath);
+          });
+
+          it(`should transform back Long UNC Windows paths (/unc/?/UNC/[server]/[sharename]/)`, () => {
+            const inputPath = `/unc/?/UNC/Server01/user/docs/Letter.txt`;
+            const outputPath = '\\\\?\\UNC\\Server01\\user\\docs\\Letter.txt';
+            expect(npath.fromPortablePath(inputPath)).toEqual(outputPath);
+          });
+
+          it(`should transform back Long UNC Windows paths (/unc/?/[drive_spec]:/)`, () => {
+            const inputPath = `/unc/?/C:/user/docs/Letter.txt`;
+            const outputPath = '\\\\?\\C:\\user\\docs\\Letter.txt';
+            expect(npath.fromPortablePath(inputPath)).toEqual(outputPath);
+          });
+
+          it(`should transform back Long UNC Windows paths with dot (/unc/./[physical_device]/)`, () => {
+            const inputPath = `/unc/./PhysicalDevice/user/docs/Letter.txt`;
```

Comment:
Hm this will be a problem too 🙁 `ppath.normalize` will transform it into `/unc/PhysicalDevice`. What does it represent?

Reply:
[This Wikipedia page](https://en.wikipedia.org/wiki/Path_(computing)#Representations_of_paths_by_operating_system_and_shell) lists it as `\\.\[physical_device]\`. That's all I know. It seems like the Node built-in `path` module also fails to normalize that kind of path, so I guess it's probably some legacy thing that nobody cares to support 😄. Going to remove the test.

Reply:
Nope, I forgot to double the backslashes 🤦‍♂️. `path.normalize` actually supports this.

Reply:
Is it OK if we store it as `/unc-dot/PhysicalDevice`?

Reply:
Sure - maybe `/unc/.dot/...`?

Reply:
🤔 But is there any requirement for server names not to be `.dot`? That could cause problems with the `\\[server]\[sharename]\` UNC paths.

Reply:
According to [this doc](https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and), valid names cannot start with a dot.

Reply:
Thanks! Done. 👍